### PR TITLE
Course management refactor.

### PR DESCRIPTION
**Pull request by gatesjm-uwm**
Refactors the course management view class to abstract out methods from GET and POST.

Also implements unit tests for all changed methods.

Includes changes from #26, accounting for refactoring, bug fixes, and back-end changes. This functionality is tested and working.

### CODE REVIEW

**Comment by LLCrafter** [comment]
Ya that looks all good to me!

**Comment by Kinneyj47** [comment]
Right now it looks like some of the methods are unimplemented, like deleteCourse and deleteSection. Is this just a github error?

**Comment by LLCrafter** [comment]
Ah I didn't catch that

File: TAScheduler/views.py [hunk 1]
```diff
@@ -155,6 +164,8 @@ def get(self, request):
         return render(request, "usermanagement.html", {"TA":TA, "Instructor":Instructor, "Admin":Admin, "Profiles":Profiles, "SideButtons":SideButtons, "UserButtons":UserButtons})
 
     def post(self, request):
+        if all([(field in request.POST) and (request.POST[field] != '') for field in
+                ['username', 'email', 'name', 'password']]):
```
Comment (LLCrafter) [inline TAScheduler/views.py:168]: This looks like its been added twice, can you double check that its only once?

**Pull request merged**